src/pgrn-custom-scan.c

```
@@ -186,31 +189,133 @@ PGrnChooseIndex(Relation table)
 	return NULL;
 }
 
+static void
+PGrnSetTargetColumns(CustomScanState *customScanState, grn_obj *targetTable)
+{
+	PGrnScanState *state = (PGrnScanState *) customScanState;
+	TupleDesc tupdesc =
+		RelationGetDescr(customScanState->ss.ss_currentRelation);
+
+	for (int i = 0; i < tupdesc->natts; i++)
+	{
+		Form_pg_attribute attr = TupleDescAttr(tupdesc, i);
+		grn_obj *column =
+			PGrnLookupColumn(targetTable, NameStr(attr->attname), ERROR);
+		GRN_PTR_PUT(ctx, &(state->columns), column);
+	}
+}
+
+static void
+PGrnSearchBuildCustomScanConditions(CustomScanState *customScanState,
+									Relation index,
+									PGrnSearchData *data)
+{
+	List *quals = customScanState->ss.ps.plan->qual;
+	ListCell *cell;
+	foreach (cell, quals)
+	{
+		Expr *expr = (Expr *) lfirst(cell);
+		OpExpr *opexpr;
+		Node *left;
+		Node *right;
+		Var *column;
+		Const *value;
+
+		if (!IsA(expr, OpExpr))
+			continue;
```

Should we report an error instead of ignore unsupported cases?